### ISSUE TITLE
Dependecy problem

### ISSUE THREAD

**Issue by vidaljessi** [soldair/node-qrcode#68]
I am using qrcode with meteor. I added the dependency using meteorhacks:npm and when I tried to use the app, I saw this error:

```
Error: Module did not self-register.
    at Error (native)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/app/programs/server/npm/npm-container/node_modules/qrcode/node_modules/canvas/lib/bindings.js:3:18)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
```
Could anybody has a clue about this issue?

**Comment by vigreco** [comment]
Hi vidaljessi,
this seems an issue with node-canvas dependency, perhaps some error occurred during module compilation?

Make sure to have installed all the needed dependencies (you can find installation instructions [here](https://github.com/Automattic/node-canvas#installation)) and try to install it again.

If you already have all the deps installed try to run `node-gyp rebuild` inside canvas folder (it should be `node_modules/qrcode/node_modules/canvas`).
If something goes wrong you should be able to see what causes the error.


_Note: node-canvas dependency will be likely removed very soon (see #67), this will make qrcode installation much easier, and will solve this kind of issues._

**Comment by vidaljessi** [comment]
These were my steps:

`sudo apt-get install libcairo2-dev libjpeg8-dev libpango1.0-dev libgif-dev build-essential g++`

`npm install canvas --save
npm install qrcode`

When I ran node-gyp rebuild  I got this:

`server/npm/npm-container/node_modules/qrcode/node_modules/canvas$ sudo node-gyp rebuild
module.js:338
    throw err;
          ^
Error: Cannot find module 'graceful-fs'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/usr/share/node-gyp/lib/node-gyp.js:12:10)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)`

**Comment by soldair** [comment]
hmm. not sure why you would have a missing sub dependency can you update npm  `sudo npm install npm -g` rm -fr node_modules and try again?

**Comment by soldair** [comment]
hey @vidaljessi we just published version 0.8.0 which npo longer has the native dependency!
please update the version string to `^0.8.0` in your npm rc and run npm install again =)

**Comment by vidaljessi** [comment]
thanks a lot, fixed!!!!!!!
